jupyter_kernel_mgmt/tests/test_client_loop.py

```diff
@@ -3,95 +3,101 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 
-
 import os
+import pytest
 try:
     from queue import Queue, Empty
 except ImportError:
     from Queue import Queue, Empty
 
 pjoin = os.path.join
-from unittest import TestCase
 
+from async_generator import yield_, async_generator
 from ipykernel.kernelspec import make_ipkernel_cmd
 from jupyter_protocol.messages import Message
-from jupyter_kernel_mgmt.subproc import SubprocessKernelLauncher
+from jupyter_kernel_mgmt.subproc.async_manager import (
+    AsyncSubprocessKernelLauncher, start_new_kernel
+)
 from jupyter_kernel_mgmt.client import ClientInThread
-from .utils import test_env
 
 from ipython_genutils.py3compat import string_types
 
 TIMEOUT = 30
 
+pytestmark = pytest.mark.asyncio
+
+@pytest.fixture
+@async_generator
```

Comment:
What Python versions do we still need to support? Async generators are properly available (i.e. without this shim) from 3.6.

Reply:
I've been assuming we need 3.5 as well.

Reply:
OK, let's leave it at 3.5 for the moment. I suspect we can drop 3.5 soon - [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html) allowed dropping it several months ago - but we don't need to think about that in this PR.